CMakePresets.json

```diff
@@ -0,0 +1,25 @@
+{
+    "version": 1,
+    "cmakeMinimumRequired": {
+        "major": 3,
+        "minor": 19,
+        "patch": 0
+    },
+    "configurePresets": [
+        {
+            "name": "osx",
+            "displayName": "osx",
+            "description": "Default build options for an OSX conda build",
+            "generator": "Ninja",
+            "binaryDir": "${sourceDir}/build",
```

Comment:
I like that this is now a way of configuring these options - it should mean we can simplify the build scripts at some point?
Should we add the `CMakeUserPresets.json` to `.gitignore` as a pre-emptive measure, as suggested at https://cmake.org/cmake/help/latest/manual/cmake-presets.7.html, to ensure it doesn't get committed?

Reply:
Also, can I override the binary directory from the command line? I know many use out-of-source builds, I do for examples as VSCode has a habit of not ignoring in-source build directories.

Reply:
Yeah hopefully this should allow us to simplify things.

You can override the build directory. If my folder structure is the following and I'm in the Documents folder, the following builds can be specified
```
/Documents/mantid
     ^
```
If I do:

`cmake mantid/ --preset=osx -B mantid_build`

It will create a build folder `/Documents/mantid_build`

And if I do:

`cmake mantid/ --preset=osx`

It will create a build folder `/Documents/mantid/build`

I like the idea of having user presets as well.